### DISCUSSION TITLE
Widget Issues

### DISCUSSION THREAD

**Discussion by inventor7777** [EdgeTX/edgetx#5282] [General]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

Hi all! I recently got a TX16S MkII Max and it’s been great! Except for some of the widgets.

They have been nothing but problems. 

Issue #1. 
When BattCheckerAnalog is placed in full screen mode, it disappears after a restart. This doesn’t seem to happen with other widgets. (The problem doesn’t happen with BattAnalog)

Issue #2
Rotary Gauge does not work. At all. I tried to set it up with all types of sources and settings, but it always stayed at -1 or 1.

Issue #3
When BattAnalog or BattChecker (analog) is placed in the top bar, its color is white when there is also a corresponding lower widget. However, it changes its color to red when to the page is changed and the battery is low, which causes problems when the top bar is red. (Even though I have the color set to white)
### Expected Behavior

I expected the widgets to stay where I left them, follow their color settings, and to show the correct values.

### Steps To Reproduce

Add BattChecker Analog to full screen layout and it disappears after restart. (See video)

Add RotaryGauge and try using it.

### Version

2.10.2

### Transmitter

RadioMaster TX16S / TX16SMK2

### Operating System (OS)

macOS

### Anything else?

If anything else is needed, let me know. Hopefully these issues can be fixed soon! (Unless I made some serious errors setting things up and there are no issues 😂)

The radio is a TX16S MKII Max 4-in-1 running a custom theme (I am hoping to open a PR for it soon!)

**Comment by pfeerick** [comment]
1) You most likely want to update the widget pack - as BattCheckerAnalog is too long a name, and IIRC was shortened to BattAnalog

2) You would need to check with the author, @offer-shmuely , but pretty sure this widget was predominantly meant for telemetry, hence why it will only work if you have a telemetry connection (and should be have big flashing "Disconnected..." text over it if it is not present. With telemetry active, it should be working with pretty much any source... i.e. it does show the elevator stick value. 

3) Lua widgets can have custom behaviours, and you are probably only configuring the *active* color for the widget... and since widgets are limited to five options, there is only so much you can configure via the UI. Check the widget source (i.e. `\WIDGETS\BattAnalog\main.lua`) and customise it to your needs.

**Comment by offer-shmuely** [comment]
Let discuss it step by step 
Where did you get widget by name BattCheckerAnalog?
We have BattCheck for the flvss of frsky
And we have BattAnalog (the name choosed carefully!) for voltage that have single value

can you give a link?

**Reply by inventor7777** [reply]
Hi!! The BattCheckerAnalog came with the radio (which was running EdgeTX 8.x when I bought it)

About why BattChecker didn’t work, that makes sense as I have a standard batt sensor.

here is the BattCheckerAnalog folder if you want to look at it:
[BattCheckAn.zip](https://github.com/user-attachments/files/16177105/BattCheckAn.zip)

I think @pfeerick is right, that the BattCheckerAnalog was just the old version of BattAnalog.

However, the rotary gauge still doesn’t work at all for me (maybe it only works with certain telemetry?)

I will update the widget pack and see if it starts to work.

**Comment by inventor7777** [comment]
I updated my SD card pack with Buddy, and it went very smoothly! 👏 

The BattCheck (Analog) vanished, and all my current set up widgets still work! 😁 

However, RotaryGauge still doesn’t not want to work. Maybe I have it set up wrong? I have it linked to my temperature telemetry sensor, set the min to 0 for testing, and max to about 160. It still just shows -1.

**Comment by offer-shmuely** [comment]
Ok, so now we are talking about the rotary
Please share the version of the script
Images of the widget, the setting and the telemetry page

**Reply by inventor7777** [reply]
I don’t know the exact version of the script, but it’s the 2.10.2 card pack. Here’s the zip just in case you need it:
[GaugeRotary.zip](https://github.com/user-attachments/files/16184500/GaugeRotary.zip)

Oddly enough, while taking screenshots I noticed that when the widget is changed to full screen (by pressing and holding), it gives an arithmetic error.
I captured a bunch of screenshots of all issues.

![IMG_4421](https://github.com/user-attachments/assets/4a8a6929-16a3-4993-b609-c8b49b7b9953)
![IMG_4422](https://github.com/user-attachments/assets/3917a38c-91c5-4fd6-94df-60837ea59a54)
![IMG_4423](https://github.com/user-attachments/assets/46cf39e8-6411-4c73-b4c5-a846340fe06c)
![IMG_4424](https://github.com/user-attachments/assets/1abdf0ce-e379-411e-88ec-f90b132cca43)
![IMG_4425](https://github.com/user-attachments/assets/039c8f00-9e4f-4ef0-9dd0-a4172d96ee5a)
![IMG_4426](https://github.com/user-attachments/assets/ee26ea1c-3e43-4c79-95dc-c0a75cdf2a8b)

**Comment by offer-shmuely** [comment]
Interesting….
Please make sure you have onle one “Temp” sensor on the telemetry page.

change the logging to “true” in the  lib_log.lua

local ENABLE_LOG_TO_FILE    = true

and send back the app.log that will be created in the widget directory

**Reply by inventor7777** [reply]
I enabled the logging, but I didn't see any log file after about a minute of testing the widget with telemetry connected (the value of the the TEMP did change a couple times during test) Maybe I have to do something different? But I did capture this screenshot of both Value and Value2 working nicely. There is only one sensor named "Temp".
![Image](https://github.com/user-attachments/assets/ed00faa5-8699-45f5-b68c-b7837523a029)

**Comment by offer-shmuely** [comment]
Can you also send this picture with and additional “value” widget that also show the “Temp” value?

**Comment by inventor7777** [comment]
I've also tried local sources like a stick/pot and couldn't get it to say anything other than -1.

**Comment by offer-shmuely** [comment]
More interesting
I do use this widget for temp & VFR on my heli
So the widget should work.
But apparently something is wrong.
Can you download load the modes using companion and send it here?

**Comment by inventor7777** [comment]
Companion gave an error (because I’m running 2.10.2 instead of 2.10.1), so here are some e compressed files that might be helpful to discovering the cause of the issue.

[Archive.zip](https://github.com/user-attachments/files/16198050/Archive.zip) (the plane I was using is model01.yml)
[radio.yml.zip](https://github.com/user-attachments/files/16198051/radio.yml.zip)

**Comment by offer-shmuely** [comment]
Can you delete all the sensors, discover them again?

**Reply by inventor7777** [reply]
That worked! I wonder why? 

Do you think you can add a “Show Min/Max” switch on the widget settings so I can hide the min and max numbers?

**Comment by offer-shmuely** [comment]
Great that it worked 😀
The edgeTx is still very limited by the setting it allow to widget.
 Part of that limitation is 5 setting per widget 
And this 5 is already occupied.
So no, not in the setting
I can add such on the file level, so it will not have gui to change, and the change will apply to all useges of the gaugeRotary on the same TX.

**Reply by inventor7777** [reply]
That would be great! 

You could perhaps offer two separate versions (like maybe GaugeRotary and GaugeRotary2)

Or just add a variable in the lua script like "minmax = yes" or something like that..

Thanks so much! and thanks for your great work on the EdgeTX widgets :)

**Comment by offer-shmuely** [comment]
found part of the issue, 
the getFieldInfo() does return nil in that model above id 322

@pfeerick can you check it?

![image](https://github.com/user-attachments/assets/44027020-5332-41df-ad87-ef9151c9b6ad)

[t_inf_bug.zip](https://github.com/user-attachments/files/16203848/t_inf_bug.zip)
[model1.zip](https://github.com/user-attachments/files/16203849/model1.zip)

**Reply by inventor7777** [reply]
That’s bizarre….
